lib/asynchronize.rb

```diff
@@ -60,15 +65,16 @@ def self.method_added(method)
   def self.create_new_method(method, klass)
     klass.instance_eval do
       old_method = instance_method(method)
-      # Can't just store the method name, since it would break if the method
-      # was redefined.
-      return if @@asynced_methods.include?(old_method)
-      undef_method(method)
-
-      @@methods_asyncing.add(method)
-      define_method(method, Asynchronize._build_new_method(old_method))
-      @@methods_asyncing.delete(method)
-      @@asynced_methods.add(instance_method(method))
+      # Old method name being stored would break if method was redefined
+      if @@asynced_methods.include?(old_method)
```

Comment:
I prefer the guard clause style here; for me it's simpler to understand, it removes a level of nesting, and it's what the [ruby style guide](https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals) recommends. 

With that said, I don't hate it.. Is there a reason you like the if..else style better here?

Reply:
I see it in several places, sometimes with several of them in a row - from flow control perspective I wonder if it's hiding another abstraction I think - for now I'm rolling it out as including it in the comments section was not intended.